### PR TITLE
Improve ext_conf_template.txt chapter

### PR DESCRIPTION
**Pull request by stmllr**
* Explain how to fetch settings
* Add example for nested settings

### CODE REVIEW

**Comment by stmllr** [comment]
Preview at: https://github.com/stmllr/TYPO3CMS-Reference-CoreApi/blob/fba94dd609497da409f2363690febaecc43b6108/Documentation/ExtensionArchitecture/ConfigurationOptions/Index.rst

File: Documentation/ExtensionArchitecture/ConfigurationOptions/Index.rst [hunk 1]
```diff
@@ -41,3 +41,27 @@ screenshot, the label – split between header and description – is
...
+      # cat=basic/enable; type=string; label=Path to the temporary directory
+      tmp = 
+      # cat=basic/enable; type=string; label=Path to the cache directory
+      cache = 
+   }
+
+This will result in  a multidimensional array::
+
+   $extensionConfiguration = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['your_extension_key']);
+   $extensionConfiguration['directories']['tmp']
```
Comment (mbrodala) [inline Documentation/ExtensionArchitecture/ConfigurationOptions/Index.rst]: This should read `$extensionConfiguration['directories.']['tmp']` (notice the dot after `directories`)

File: Documentation/ExtensionArchitecture/ConfigurationOptions/Index.rst [hunk 2]
```diff
@@ -41,3 +41,27 @@ screenshot, the label – split between header and description – is
...
+   $extensionConfiguration['directories']['cache']
```
Comment (mbrodala) [inline Documentation/ExtensionArchitecture/ConfigurationOptions/Index.rst]: Same here.

File: Documentation/ExtensionArchitecture/ConfigurationOptions/Index.rst (hunk 1 above)
Comment (stmllr) [inline Documentation/ExtensionArchitecture/ConfigurationOptions/Index.rst]: fixed

File: Documentation/ExtensionArchitecture/ConfigurationOptions/Index.rst (hunk 2 above)
Comment (stmllr) [inline Documentation/ExtensionArchitecture/ConfigurationOptions/Index.rst]: fixed

File: Documentation/ExtensionArchitecture/ConfigurationOptions/Index.rst [hunk 3]
```diff
@@ -63,5 +63,11 @@ You can also define nested options using the TypoScript notation::
...
    $extensionConfiguration = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['your_extension_key']);
-   $extensionConfiguration['directories']['tmp']
-   $extensionConfiguration['directories']['cache']
+   $extensionConfiguration['directories.']['tmp']
+   $extensionConfiguration['directories.']['cache']
+
+.. important::
+
+   Don't forget the dot at the end of the array key :code:`directories`.
+   This is a convention from the TypoScript parser.
```
Comment (mbrodala) [inline Documentation/ExtensionArchitecture/ConfigurationOptions/Index.rst]: Suggestion: "Notice the dot at the end of the `directories` key. This notation must be used for every grouping key and is a convention of the TypoScript parser."

**Comment by stmllr** [comment]
This PR should also be backported to 6-2 and 7-6.

**Pull request merged**